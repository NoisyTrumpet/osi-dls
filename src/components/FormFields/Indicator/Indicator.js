import React from 'react';
import PropTypes from 'prop-types';
import './Indicator.scss';
import classNames from 'classnames';

const Indicator = ({
  children,
  type,
  error,
}) => {
  const classes = classNames(
    'indicator',
    error,
    `indicator-${type}`,
  );

  return (
    <span className={classes} aria-hidden="true">
      {children}
    </span>
  );
};

Indicator.defaultProps = {
  children: '',
  type: undefined,
};

Indicator.propTypes = {
  children: PropTypes.node,
  type: PropTypes.oneOf([
    'radio',
    'checkbox',
    'toggle',
    'hidden',
  ]),
  error: PropTypes.bool,
};

export default Indicator;
