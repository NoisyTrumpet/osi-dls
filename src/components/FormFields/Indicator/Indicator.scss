.indicator {
  background-color: map-get($colors, 'brand-transparent');
  border: rem(2px) solid map-get($colors, 'primary');
  height: rem(18px);
  width: rem(18px);
  transition: $timing-0;

  &:after {
    content: "";
    opacity: 0;
    position: absolute;
    transition: inherit;
  }

  &-radio {
    border-radius: 50%;
  }

  &-checkbox {
    border-radius: $space-4;
  }
}

// checkbox
.input-field-checkbox {

  .indicator:after {
    border: solid map-get($colors, 'brand-white');
    border-width: 0 rem(2px) rem(2px) 0;
    height: rem(10px);
    width: rem(6px);
    top: rem(4px);
    left: rem(5px);
    transform: rotate(45deg);
  }

}

// radio
.input-field-radio {

  .indicator:after {
    background: map-get($colors, 'primary');
    border: solid rem(2px) map-get($colors, 'brand-white');
    border-radius: 50%;
    height: rem(10px);
    width: rem(10px);
    top: rem(5px);
    left: rem(2px);
  }

}

// toggle
.indicator-toggle {
  position: absolute;
  cursor: pointer;
  margin-left: $space-12;
  height: $space-24;
  width: $space-48;
  background-color: map-get($colors, 'brand-transparent');
  -webkit-transition: $timing-1;
  transition: $timing-1;
  border-radius: $space-12;
  border: rem(2px) solid map-get($colors, 'primary');

  &:before {
    position: absolute;
    content: "";
    height: $space-16;
    width: $space-16;
    left: rem(2px);
    bottom: rem(2px);
    background-color: map-get($colors, 'primary');
    -webkit-transition: $timing-1;
    transition: $timing-1;
    border-radius: 50%;
  }
}

.input-field:checked + .indicator-toggle {
  background-color: map-get($colors, 'primary');

  &:before {
    background-color: map-get($colors, 'gray1');
    -webkit-transform: translateX($space-24);
    -ms-transform: translateX($space-24);
    transform: translateX($space-24);
  }
}

.input-field:focus + .indicator-toggle {
  box-shadow: 0 0 rem(1px) map-get($colors, 'accentB1');
}

.indicator-hidden {
  position: fixed;
  opacity: 0;
}