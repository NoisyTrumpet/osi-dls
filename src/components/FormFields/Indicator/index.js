export { default as Indicator } from './Indicator'; // eslint-disable-line import/prefer-default-export
