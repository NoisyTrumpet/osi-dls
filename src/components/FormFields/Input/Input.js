import React from 'react';
import PropTypes from 'prop-types';
import './Input.scss';
import classNames from 'classnames';

const Input = ({
  ariaControls,
  autoComplete,
  click,
  domId,
  error,
  isChecked,
  isDisabled,
  maxLength,
  name,
  onBlur,
  onChange,
  onFocus,
  placeholder,
  required,
  type,
  value,
  // range input
  min,
  max,
  step,
}) => {
  const classes = classNames(
    'input-field',
    { error },
  );

  return (
    <input
      aria-controls={ariaControls}
      autoComplete={autoComplete || undefined}
      type={type}
      name={name}
      id={domId}
      className={classes}
      maxLength={maxLength}
      placeholder={placeholder}
      defaultValue={value}
      checked={isChecked}
      disabled={isDisabled}
      aria-required={required}
      aria-invalid={error}
      onChange={onChange}
      onBlur={onBlur}
      onClick={click}
      onFocus={onFocus}
      min={min}
      max={max}
      step={step}
    />
  );
};

Input.defaultProps = {
  domId: '',
  name: '',
  onChange: null,
  onBlur: null,
  placeholder: '',
  type: 'text',
  value: '',
};

Input.propTypes = {
  ariaControls: PropTypes.string,
  autoComplete: PropTypes.oneOf(['on', 'off']),
  click: PropTypes.func,
  domId: PropTypes.string,
  error: PropTypes.bool,
  isChecked: PropTypes.bool,
  isDisabled: PropTypes.bool,
  maxLength: PropTypes.number,
  name: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  type: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]),
  min: PropTypes.string,
  max: PropTypes.string,
  step: PropTypes.string,
};

export default Input;
