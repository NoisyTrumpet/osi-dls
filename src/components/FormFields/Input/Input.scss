.input-field {
  color: map-get($colors, 'brand-grey');
  border: rem(2px) solid map-get($colors, 'gray3');
  border-radius: $space-8;
  order: 1;
  padding: $space-12;
  @include fontSans($beam-84, $beam-vf, 'sans-serif', 84, 700);

  &:focus {
    border-color: map-get($colors, 'brand-blue');
    outline: none;
  }

  &[type='checkbox'],
  &[type='radio'] {
    cursor: pointer;
    opacity: 0;
    position: absolute;
    z-index: 1;
    @include appearance(none);
  }

  &[type='checkbox'],
  &[type='radio'] {
    height: rem(18px);
    width: rem(18px);
  }

  &[type='text'] {
    width: 100%;
  }

  &:disabled {
    color: map-get($colors, 'disabled');
    cursor: default;
    border-color: map-get($colors, 'disabled');

    ~ .input-label {
      cursor: default;

      .typography {
        color: map-get($colors, 'disabled');
      }
    }

    ~ .indicator {
      border-color: map-get($colors, 'disabled');

      &:after {
        background-color: map-get($colors, 'disabled');
      }
    }
  }

  &.error {
    border-color: map-get($colors, 'util-error');

    &:focus {
      border-color: map-get($colors, 'util-error');
    }
  }
}

.input-field-text {
  .helper-text {
    padding-left: 15px;
  }
}

.input-field-select {
  .helper-text {
    padding-left: 15px;
  }
}

// container ... normally goes in a forms.scss
.input-field-checkbox,
.input-field-radio {
  align-items: center;
  display: flex;
  position: relative;

  .input-field,
  .input-label {
    &:hover {
      ~ .indicator {
        background-color: map-get($colors, 'gray1');
      }
    }
  }

  .input-field {
    &:checked {
      ~ .indicator {
        background-color: map-get($colors, 'brand-blue');

        &:after {
          opacity: 1;
        }
      }
    }

    &:disabled {
      ~ .indicator {
        background-color: map-get($colors, 'brand-transparent');
      }
    }
  }

  .input-label {
    cursor: pointer;
    line-height: $space-24;
    @include user-select;

    &.isDisabled {
      cursor: auto;
    }
  }
}

.input-field-checkbox {
  .helper-text {
    margin: 0 auto 6px -26px;
  }
}

// Prevent visual toggle when helpertext appears
.no-helpertext {
  margin-bottom: 30px;
}
