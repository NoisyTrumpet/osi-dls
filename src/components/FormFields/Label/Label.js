import React from 'react';
import PropTypes from 'prop-types';
import './Label.scss';
import classNames from 'classnames';
import { Typography } from '../../Typography';

const Label = ({
  fieldType,
  forId,
  display,
  hideLabel,
  isDisabled,
  color,
  required,
  addClass,
  suffixHelperText,
  children,
}) => {
  const classes = classNames(
    'input-label',
    fieldType,
    display,
    hideLabel && 'hide-label',
    addClass,
    { isDisabled },
  );

  const srtVariant = hideLabel && "srt";

  return (
    <label
      className={classes}
      htmlFor={forId}
    >
      <Typography variant={srtVariant || "field-label"} color={color} addClass={addClass}>
        {children}
      </Typography>
      {suffixHelperText && (
        <Typography variant="helper" addClass="optional">
          {suffixHelperText}
        </Typography>
      )}
      {!required && (
        <Typography variant={srtVariant || "helper"} addClass="optional">
          (Optional)
        </Typography>
      )}
    </label>
  );
};

Label.defaultProps = {
  fieldType: 'text',
  display: 'inline',
  isDisabled: false,
  color: '',
};

Label.propTypes = {
  fieldType: PropTypes.oneOf([
    'text',
    'radio',
    'checkbox',
    'textarea',
    'select',
  ]),
  forId: PropTypes.string,
  display: PropTypes.oneOf([
    'inline',
    'inline-block',
    'block',
  ]),
  hideLabel: PropTypes.bool,
  isDisabled: PropTypes.bool,
  color: PropTypes.string,
  suffixHelperText: PropTypes.string, // helper text that is appended to label
  required: PropTypes.bool,
  addClass: PropTypes.string,
  children: PropTypes.node,
};

export default Label;
