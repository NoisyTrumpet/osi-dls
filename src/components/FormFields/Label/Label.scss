.input-label {
  color: map-get($colors, 'gray5');
  padding-left: $space-8;
  order: 2;

  &.block {
    display: block;
  }

  &.inline-block {
    display: inline-block;
  }

  &.text {
    position: absolute;
    top: rem(-11px);
    left: rem(2px);
    transform: translate(0, 50%);

    .optional {
      font-size: rem(12px);
      padding-left: rem(2px);
      padding-right: 0px;
      position: absolute;
      background: #ffffff;
    }
    
    .typography-field-label {
      background: map-get($colors, 'brand-white');
      padding: 0 $space-4;
    }
  }
}
