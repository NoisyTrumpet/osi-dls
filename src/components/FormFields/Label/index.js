export { default as Label } from './Label'; // eslint-disable-line import/prefer-default-export
