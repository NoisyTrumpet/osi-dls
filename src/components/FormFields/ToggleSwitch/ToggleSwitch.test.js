import React from 'react';
import { mount } from 'enzyme';
import { Formik, Form } from 'formik';
import toJson from 'enzyme-to-json';
import { ToggleSwitch } from './';
import { Indicator } from '../Indicator';
import { Input } from '../Input';
import { Label } from '../Label';

describe('<ToggleSwitch />', () => {
  test('should render ', () => {
    const text = 'Toggle Switch';
    const domId = 'toggleSwitch';
    const checked = false;
    const wrapper = mount(
      <Formik initialValues={{ toggle: '' }}>
        <Form>
          <ToggleSwitch
            name="toggle"
            domId={domId}
            label={text}
            checked={checked}
          />
        </Form>
      </Formik>,
    );
    expect(wrapper.find(Label).length).toBe(1);
    expect(wrapper.find(Input).length).toBe(1);
    expect(wrapper.find(Indicator).length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
