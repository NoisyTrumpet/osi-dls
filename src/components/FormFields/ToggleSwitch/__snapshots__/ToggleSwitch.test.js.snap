// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> should render  1`] = `
<Formik
  initialValues={
    Object {
      "toggle": "",
    }
  }
>
  <Form>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <ToggleSwitch
        checked={false}
        domId="toggleSwitch"
        label="Toggle Switch"
        labelPosition="left"
        name="toggle"
      >
        <div
          className="toggle-switch no-helpertext"
        >
          <Label
            color=""
            display="inline"
            fieldType="checkbox"
            forId="toggleSwitch"
            isDisabled={false}
            required={true}
          >
            <label
              className="input-label checkbox inline"
              htmlFor="toggleSwitch"
            >
              <Typography
                color=""
                variant="field-label"
              >
                <span
                  className="typography typography-field-label"
                >
                  Toggle Switch
                </span>
              </Typography>
            </label>
          </Label>
          <Input
            checked={false}
            domId="toggleSwitch"
            error={false}
            isChecked={false}
            name="toggle"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="checkbox"
            value=""
          >
            <input
              aria-invalid={false}
              checked={false}
              className="input-field"
              defaultValue=""
              id="toggleSwitch"
              name="toggle"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="checkbox"
            />
          </Input>
          <Indicator
            type="toggle"
          >
            <span
              aria-hidden="true"
              className="indicator indicator-toggle"
            />
          </Indicator>
        </div>
      </ToggleSwitch>
    </form>
  </Form>
</Formik>
`;
